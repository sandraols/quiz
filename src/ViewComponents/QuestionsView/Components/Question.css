.Question {
  width: 40vw;
}

.question__container {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.question__paragraph {
  margin-bottom: 2rem;
}

.question__gif-container {
  width: 50vw;
  height: 40vh;
}

.question__giphy-embed {
  pointer-events: none;
  width: 100%;
  height: 30vh;
}

.question__alternatives-container {
  position: absolute;
  bottom: 10vh;
  width: 50vw;
  display: flex;
  flex-direction: column;
  justify-content: center;
}

.question__alternatives-top,
.question__alternatives-bottom {
  display: flex;
  justify-content: center;
}