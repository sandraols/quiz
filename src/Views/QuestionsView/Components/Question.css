.Question {
  width: 60vw;
  height: 80vh;
}

.question__container {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.question__gif-container {
  width: 50vw;
  height: 40vh;
}

.question__giphy-embed {
  pointer-events: none;
  width: 31rem;
  height: 18rem
}

.question__alternative-container {
  position: absolute;
  bottom: 30vh;
}