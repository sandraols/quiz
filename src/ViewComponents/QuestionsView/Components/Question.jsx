import { React } from 'react';
import './Question.css';

export const Question = (
  { 
    question, 
    index, 
    nextQuestion, 
    increaseScore, 
    increaseNumOfIncorrectAnswers, 
    twoFalseAlternatives
  }
) => {

  const questionAlternatives = question.alternatives;
  const gifLink = question.gif.gifLink;
  const gifLinkEmbedded = question.gif.gifLinkEmbedded;
  const gifTitle = question.gif.title;

  const checkIfCorrect = (index) => {
    increaseScore(questionAlternatives[index].isCorrect);
    increaseNumOfIncorrectAnswers(questionAlternatives[index].isCorrect);
  }

  const handleClick = (index) => {
    checkIfCorrect(index);
    nextQuestion();
  }

  return(
    <section className="Question">
      <div className="question__container">
      <h1>Question {index}</h1>
        <p className="question__paragraph">
          {question.question}
        </p>
        <div className="question__gif-container">
          <iframe 
            src={gifLinkEmbedded}
            title={gifTitle}  
            frameBorder="0" 
            className="question__giphy-embed" 
          ></iframe>
          <p>
            <a href={gifLink} rel="noreferrer" target="_blank">via GIPHY</a>
          </p>
        </div>
        <div className="question__alternatives-container">
          <div className="question__alternatives-top">
            <button
              onClick={() => {handleClick(0)}}
              disabled={twoFalseAlternatives.includes(questionAlternatives[0])}
            >
              {questionAlternatives[0].alternative}
            </button>
            <button 
              onClick={() => {handleClick(1)}} 
              disabled={twoFalseAlternatives.includes(questionAlternatives[1])}
            >
              {questionAlternatives[1].alternative}
            </button>
          </div>
          <div className="question__alternatives-bottom">
            <button 
              onClick={() => {handleClick(2)}}
              disabled={twoFalseAlternatives.includes(questionAlternatives[2])} 
            >
              {questionAlternatives[2].alternative}
            </button>
            <button 
              onClick={() => {handleClick(3)}}
              disabled={twoFalseAlternatives.includes(questionAlternatives[3])} 
            >
              {questionAlternatives[3].alternative}
            </button>
          </div>
        </div>
      </div>
    </section>
  )
}