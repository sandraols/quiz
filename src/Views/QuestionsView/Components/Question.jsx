import { React } from 'react';
import './Question.css';

export const Question = ({question, index, nextQuestion, increaseScore}) => {

  const questionAlternatives = question.alternatives;
  const gifLink = question.gif.gifLink;
  const gifLinkEmbedded = question.gif.gifLinkEmbedded;
  const gifTitle = question.gif.title;

  const checkIfCorrect = (index) => {
    increaseScore(questionAlternatives[index].isCorrect);
  }

  const handleClick = (index) => {
    checkIfCorrect(index);
    nextQuestion();
  }

  return(
    <section className="Question">
      <div className="question__container">
      <h1 className="question__title">Question {index}</h1>
        <p className="question__question">
          {question.question}
        </p>
        <div className="question__gif-container">
          <iframe 
            src={gifLinkEmbedded}
            title={gifTitle}  
            frameBorder="0" 
            className="question__giphy-embed" 
          ></iframe>
          <p>
            <a href={gifLink} rel="noreferrer" target="_blank">via GIPHY</a>
          </p>
        </div>
        <div className="question__alternative-container">
          <button
            onClick={() => {handleClick(0)}}
            className="question__alternative"
          >
              {questionAlternatives[0].alternative}
          </button>
          <button 
            onClick={() => {handleClick(1)}} 
            className="question__alternative"
          >
            {questionAlternatives[1].alternative}
          </button>
          <button 
            onClick={() => {handleClick(2)}} 
            className="question__alternative"
          >
            {questionAlternatives[2].alternative}
          </button>
          <button 
            onClick={() => {handleClick(3)}} 
            className="question__alternative"
          >
            {questionAlternatives[3].alternative}
          </button>
        </div>
        {/* <button className="question__next" onClick={nextQuestion}>Next question</button> */}
      </div>
    </section>
  )
}