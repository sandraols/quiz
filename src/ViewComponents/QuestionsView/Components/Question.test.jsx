import React from 'react';
import { shallow } from 'enzyme';
import { Question } from './Question';

describe('IntroView', () => {
  let wrapper;
  const question = {
    question: "The question",
    alternatives: [
      {
        alternative: "one",
        isCorrect: true
      },
      {
        alternative: "two",
        isCorrect: false
      },
      {
        alternative: "three",
        isCorrect: false
      },
      {
        alternative: "four",
        isCorrect: false
      }
    ],
    gif: {
      gifLink: "a link",
      gifLinkEmbedded: "another link",
      title: "a title"
    }
  }
  const index = 1;
  const nextQuestionMock = jest.fn();
  const increaseScoreMock = jest.fn();
  const increaseNumOfIncorrectAnswersMock = jest.fn();
  const twoFalseAlternatives = ['false alternative one', 'false alternative two']

  beforeEach(() => {
    wrapper = shallow(
      <Question 
        question={question} 
        index={index}  
        nextQuestion={nextQuestionMock}
        increaseScore={increaseScoreMock} 
        increaseNumOfIncorrectAnswers={increaseNumOfIncorrectAnswersMock}
        twoFalseAlternatives={twoFalseAlternatives}
      />
    );
  });
  
  it('renders without crashing', () => {
    console.log(wrapper.debug());
    expect(wrapper).toMatchSnapshot();
  });

  describe('h1 element', () => {
    it('should render "Question" and which question it is currently at', () => {
      wrapper.find('h1').text().includes(`Question ${index}`);
    });
  });

  describe('.question__paragraph', () => {
    it('should render a question', () => {
      wrapper.find('.question__paragraph').text().includes(question.question);
    });
  });

  describe('alternative buttons', () => {
    it('should not have called nextQuestion by default', () => {
      expect(nextQuestionMock).not.toHaveBeenCalled();
    });
    describe('when a correct alternative button is clicked', () => {
      it('should call nextQuestion', () => {
        wrapper.find('button').at(0).simulate('click');
        expect(nextQuestionMock).toHaveBeenCalledTimes(1);
      });
      it('should call increaseScore', () => {
        wrapper.find('button').at(0).simulate('click');
        expect(increaseScoreMock).toHaveBeenCalledTimes(1);
      });
    });
    describe('when an incorrect alternative button is clicked', () => {
      it('should call nextQuestion', () => {
        wrapper.find('button').at(1).simulate('click');
        expect(nextQuestionMock).toHaveBeenCalledTimes(1);
      });
      it('should call increaseNumOfIncorrectAnswers', () => {
        wrapper.find('button').at(0).simulate('click');
        expect(increaseNumOfIncorrectAnswersMock).toHaveBeenCalledTimes(1);
      });
    });
  });
});