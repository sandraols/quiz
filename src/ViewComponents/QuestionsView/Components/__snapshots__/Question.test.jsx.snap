// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IntroView renders without crashing 1`] = `ShallowWrapper {}`;
